### ISSUE TITLE
Having installed from .deb file, the DriveShare GUI prompts to install from .deb file (probably because of dependencies not being installed properly)

### ISSUE THREAD

**Issue by pryds** [Storj/farmer-gui#196]
On a fresh install of Ubuntu 16.04 LTS, I downloaded driveshare_0.5.1.amd64.deb and installed it without any errors using:
```
sudo dpkg -i driveshare_0.5.1.amd64.deb
```
Then installed dependencies using:
```
sudo apt-get install -f
```

The latter command resulted in the following error:
```
$ sudo apt-get install -f
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Correcting dependencies... Done
The following additional packages will be installed:
  libexpat1-dev libpython-all-dev libpython-dev libpython2.7-dev python-all
  python-all-dev python-dev python-pip python-pip-whl python-wheel
  python2.7-dev
Suggested packages:
  python-setuptools
The following NEW packages will be installed:
  libexpat1-dev libpython-all-dev libpython-dev libpython2.7-dev python-all
  python-all-dev python-dev python-pip python-pip-whl python-wheel
  python2.7-dev
0 upgraded, 11 newly installed, 0 to remove and 10 not upgraded.
1 not fully installed or removed.
Need to get 0 B/29.4 MB of archives.
After this operation, 44.2 MB of additional disk space will be used.
Do you want to continue? [Y/n] 
Selecting previously unselected package python-pip-whl.
(Reading database ... 210988 files and directories currently installed.)
Preparing to unpack .../python-pip-whl_8.1.1-1_all.deb ...
Unpacking python-pip-whl (8.1.1-1) ...
Selecting previously unselected package python-pip.
Preparing to unpack .../python-pip_8.1.1-1_all.deb ...
Unpacking python-pip (8.1.1-1) ...
Selecting previously unselected package libexpat1-dev:amd64.
Preparing to unpack .../libexpat1-dev_2.1.0-7_amd64.deb ...
Unpacking libexpat1-dev:amd64 (2.1.0-7) ...
Selecting previously unselected package libpython2.7-dev:amd64.
Preparing to unpack .../libpython2.7-dev_2.7.11-7_amd64.deb ...
Unpacking libpython2.7-dev:amd64 (2.7.11-7) ...
Selecting previously unselected package libpython-dev:amd64.
Preparing to unpack .../libpython-dev_2.7.11-1_amd64.deb ...
Unpacking libpython-dev:amd64 (2.7.11-1) ...
Selecting previously unselected package python2.7-dev.
Preparing to unpack .../python2.7-dev_2.7.11-7_amd64.deb ...
Unpacking python2.7-dev (2.7.11-7) ...
Selecting previously unselected package python-dev.
Preparing to unpack .../python-dev_2.7.11-1_amd64.deb ...
Unpacking python-dev (2.7.11-1) ...
Selecting previously unselected package libpython-all-dev:amd64.
Preparing to unpack .../libpython-all-dev_2.7.11-1_amd64.deb ...
Unpacking libpython-all-dev:amd64 (2.7.11-1) ...
Selecting previously unselected package python-all.
Preparing to unpack .../python-all_2.7.11-1_amd64.deb ...
Unpacking python-all (2.7.11-1) ...
Selecting previously unselected package python-all-dev.
Preparing to unpack .../python-all-dev_2.7.11-1_amd64.deb ...
Unpacking python-all-dev (2.7.11-1) ...
Selecting previously unselected package python-wheel.
Preparing to unpack .../python-wheel_0.29.0-1_all.deb ...
Unpacking python-wheel (0.29.0-1) ...
Processing triggers for man-db (2.7.5-1) ...
Processing triggers for doc-base (0.10.7) ...
Processing 1 added doc-base file...
Registering documents with scrollkeeper...
Setting up python-pip-whl (8.1.1-1) ...
Setting up python-pip (8.1.1-1) ...
Setting up libexpat1-dev:amd64 (2.1.0-7) ...
Setting up libpython2.7-dev:amd64 (2.7.11-7) ...
Setting up libpython-dev:amd64 (2.7.11-1) ...
Setting up python2.7-dev (2.7.11-7) ...
Setting up python-dev (2.7.11-1) ...
Setting up driveshare (0.5.1) ...
The directory '/home/thomas/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
The directory '/home/thomas/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Collecting dataserv-client
  Downloading dataserv_client-2.1.12-py2-none-any.whl
Collecting btctxstore>=4.6.0 (from dataserv-client)
  Downloading btctxstore-4.6.1.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
    ImportError: No module named setuptools
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-Q5qjra/btctxstore/
ln: failed to create symbolic link '/usr/local/bin/driveshare': File exists
dpkg: error processing package driveshare (--configure):
 subprocess installed post-installation script returned error exit status 1
Setting up libpython-all-dev:amd64 (2.7.11-1) ...
Setting up python-all (2.7.11-1) ...
Setting up python-all-dev (2.7.11-1) ...
Setting up python-wheel (0.29.0-1) ...
Errors were encountered while processing:
 driveshare
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

Now, when I start DriveShare from the GUI, I get an error message: "It looks like you are missing the required dependencies for GNU/Linux. Please install DriveShare using our Debian package. When I click the "Download" button (only available choice), I am taken to the website where I downloaded the DriveShare deb package in the first place.

![driveshare](https://cloud.githubusercontent.com/assets/4420584/14364371/d1fb25d4-fd08-11e5-8fe9-69c7c069f163.png)

**Comment by littleskunk** [comment]
You should run into this issue: https://github.com/Storj/farmer-gui/issues/182

Your error message looks different but please try the workaround.

**Comment by littleskunk** [comment]
Hold on. I remember what the problem is. Your pip version is to old. Please upgrade the pip version.

`easy_install --upgrade pip`

**Comment by gordonwritescode** [comment]
This will no longer be an issue with 0.6.0

**Comment by pryds** [comment]
> Hold on. I remember what the problem is. Your pip version is to old. Please upgrade the pip version.

Ok, easy_install doesn't seem to be among the dependencies of the deb file, so I did `sudo apt-get install python-setuptools` and then `sudo easy_install --upgrade pip` which upgraded pip to 8.1.1. Then I did `sudo apt-get remove driveshare` and tried installing it again, which gave the following error:
```
$ sudo dpkg -i driveshare_0.5.1.amd64.deb 
Selecting previously unselected package driveshare.
(Reading database ... 212673 files and directories currently installed.)
Preparing to unpack driveshare_0.5.1.amd64.deb ...
Unpacking driveshare (0.5.1) ...
Setting up driveshare (0.5.1) ...
The directory '/home/thomas/.cache/pip/http' or its parent directory is not owned by the current user and the cache has been disabled. Please check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
The directory '/home/thomas/.cache/pip' or its parent directory is not owned by the current user and caching wheels has been disabled. check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Requirement already satisfied (use --upgrade to upgrade): dataserv-client in /usr/local/lib/python2.7/dist-packages
Requirement already satisfied (use --upgrade to upgrade): btctxstore>=4.6.0 in /usr/local/lib/python2.7/dist-packages (from dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): partialhash>=1.1.0 in /usr/local/lib/python2.7/dist-packages (from dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): storjcore>=0.0.6 in /usr/local/lib/python2.7/dist-packages (from dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): future>=0.15.0 in /usr/local/lib/python2.7/dist-packages (from dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): RandomIO>=0.2.1 in /usr/local/lib/python2.7/dist-packages (from dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): storjnode>=0.0.18 in /usr/local/lib/python2.7/dist-packages (from dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): psutil>=3.2.2 in /usr/local/lib/python2.7/dist-packages (from dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): pycoin>=0.52 in /usr/local/lib/python2.7/dist-packages (from btctxstore>=4.6.0->dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): six>=1.10.0 in /usr/lib/python2.7/dist-packages (from btctxstore>=4.6.0->dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): ecdsa>=0.13 in /usr/local/lib/python2.7/dist-packages (from btctxstore>=4.6.0->dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): pycrypto>=2.6.1 in /usr/local/lib/python2.7/dist-packages (from storjcore>=0.0.6->dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): crochet>=1.4.0 in /usr/local/lib/python2.7/dist-packages (from storjnode>=0.0.18->dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): netifaces>=0.10.4 in /usr/local/lib/python2.7/dist-packages (from storjnode>=0.0.18->dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): twisted>=15.4.0 in /usr/local/lib/python2.7/dist-packages (from storjnode>=0.0.18->dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): pyp2p>=0.4.5 in /usr/local/lib/python2.7/dist-packages (from storjnode>=0.0.18->dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): ipaddress>=1.0.14 in /usr/lib/python2.7/dist-packages (from storjnode>=0.0.18->dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): kademlia>=0.5 in /usr/local/lib/python2.7/dist-packages (from storjnode>=0.0.18->dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): requests>=2.8.1 in /usr/local/lib/python2.7/dist-packages (from storjnode>=0.0.18->dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): ntplib>=0.3.3 in /usr/local/lib/python2.7/dist-packages (from storjnode>=0.0.18->dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): zope.interface>=3.6.0 in /usr/local/lib/python2.7/dist-packages (from twisted>=15.4.0->storjnode>=0.0.18->dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): pyroute2>=0.3.15 in /usr/local/lib/python2.7/dist-packages (from pyp2p>=0.4.5->storjnode>=0.0.18->dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): rpcudp>=1.0 in /usr/local/lib/python2.7/dist-packages (from kademlia>=0.5->storjnode>=0.0.18->dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): setuptools in /usr/lib/python2.7/dist-packages (from zope.interface>=3.6.0->twisted>=15.4.0->storjnode>=0.0.18->dataserv-client)
Requirement already satisfied (use --upgrade to upgrade): u-msgpack-python>=1.5 in /usr/local/lib/python2.7/dist-packages (from rpcudp>=1.0->kademlia>=0.5->storjnode>=0.0.18->dataserv-client)
ln: failed to create symbolic link '/usr/local/bin/driveshare': File exists
dpkg: error processing package driveshare (--install):
 subprocess installed post-installation script returned error exit status 1
Processing triggers for desktop-file-utils (0.22-1ubuntu5) ...
Processing triggers for bamfdaemon (0.5.3~bzr0+16.04.20160331-0ubuntu1) ...
Rebuilding /usr/share/applications/bamf-2.index...
Processing triggers for gnome-menus (3.13.3-6ubuntu3) ...
Processing triggers for mime-support (3.59ubuntu1) ...
Errors were encountered while processing:
 driveshare
```

**Comment by littleskunk** [comment]
`rm /usr/local/bin/driveshare` and try again

**Comment by pryds** [comment]
Seems to work, thanks. Sounds great that this will be fixed in 0.6.0
